libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts

```diff
@@ -1455,17 +1516,28 @@ const ParentalLeaveTemplate: ApplicationTemplate<
 
         return context
       }),
-      setHasApplied: assign((context) => {
+      setDateOfBirth: assign((context) => {
         const { application } = context
         const { answers } = application
-        if (
-          answers.previousState === 'approved' &&
-          application.state === 'vinnumalastofnunApproveEdits'
-        ) {
-          set(answers, 'hasAppliedFor', 'residenceGrant')
+        const data = (application.externalData.dateOfBirth.data as unknown) as {
```

Comment:
I think you should add this to the getApplicationExternalData in parentalLeaveUtils, something like this:
const dateOfBirth = getValueViaPath(externalData, 'dateOfBirth') as string